src/main/java/ca/uhn/fhir/jpa/starter/CustomConsentService.java

```diff
@@ -0,0 +1,108 @@
+package ca.uhn.fhir.jpa.starter;
+
+import java.util.List;
+import org.hl7.fhir.instance.model.api.IBaseResource;
+import org.hl7.fhir.r4.model.Base;
+import org.hl7.fhir.r4.model.Patient;
+import org.hl7.fhir.r4.model.Reference;
+import org.hl7.fhir.r4.model.Resource;
+import org.springframework.stereotype.Service;
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import ca.uhn.fhir.rest.api.RequestTypeEnum;
+import ca.uhn.fhir.rest.api.server.RequestDetails;
+import ca.uhn.fhir.rest.server.interceptor.consent.ConsentOutcome;
+import ca.uhn.fhir.rest.server.interceptor.consent.IConsentContextServices;
+import ca.uhn.fhir.rest.server.interceptor.consent.IConsentService;
+
+@Service
+public class CustomConsentService implements IConsentService {
+
+  private static final String OAUTH_CLAIM_NAME = System.getenv("OAUTH_CLAIM_NAME");
+
+  private OAuth2Helper oAuth2Helper = new OAuth2Helper();
+
+  @Override
+  public ConsentOutcome startOperation(RequestDetails theRequestDetails,
+      IConsentContextServices theContextServices) {
+    RequestTypeEnum requestType = theRequestDetails.getRequestType();
+    if (requestType.equals(RequestTypeEnum.POST) || requestType.equals(RequestTypeEnum.PUT)) {
+      return validateRequest(theRequestDetails);
+    } else if (requestType.equals(RequestTypeEnum.GET)
+        || requestType.equals(RequestTypeEnum.DELETE)) {
+      return ConsentOutcome.PROCEED;
+    }
+    return ConsentOutcome.REJECT;
+  }
+
+  @Override
+  public ConsentOutcome canSeeResource(RequestDetails theRequestDetails, IBaseResource theResource,
+      IConsentContextServices theContextServices) {
+    return ConsentOutcome.PROCEED;
```

Comment:
The comment regarding performance implications in the example should probably be kept here for others to see.

```
// This means that all requests should flow through the consent service
// This has performance implications - If you know that some requests
// don't need consent checking it is a good idea to return
// ConsentOutcome.AUTHORIZED instead for those requests.
```

Reply:
Should we test for the patient claim here? The comment in the example (that I suggested adding in the comment above) says it's a good idea to return `ConsentOutcome.AUTHORIZED` when consent checking isn't needed.